gemini/use-cases/retrieval-augmented-generation/qna_using_query_routing/utils/intent_routing.py

```diff
@@ -0,0 +1,413 @@
+            temperature=float(self.config["genai_chat"]["temperature"]),
+            max_output_tokens=int(self.config["genai_chat"]["max_output_tokens"]),
+        )
+
+    def greetings(self, text: str) -> str:
+        """
+        Generates a friendly greeting message in response to a 'WELCOME' intent.
+
+        Leverages the provided chat model to generate a greeting tailored to the enabled programming languages.
+
+        Args:
+            text (str): The user's original message.
+
+        Returns:
+            str: The generated greeting message.
+        """
+
+        enabled_programming_language_list = self.config["default"][
+            "enabled_programming_language"
+        ].split(",")
+        enabled_programming_language = ", ".join(
+            [lang.title() for lang in enabled_programming_language_list]
+        )
+
+        chat = self.model.start_chat()
+        response = chat.send_message(
+            f"""You are Generative AI powered genai Learning Assistant.
+
+        You are trained ONLY to answers questions related to following programming languages: {enabled_programming_language}
+
+        Write a brief greeting message:"""
+        )
+        parameters_local = GenerationConfig(
+            temperature=0.7,
+            max_output_tokens=int(self.config["genai_chat"]["max_output_tokens"]),
+        )
+
+        response = chat.send_message(f"""{text}""", generation_config=parameters_local)
+        message = response.text
+        return message
+
+    def closing(self, text: str) -> str:
+        """
+        Generates a closing/thank you message in response to a 'CLOSE' intent.
+
+        Leverages the provided chat model to generate a closing message.
+
+        Args:
+            text (str): The user's original message.
+
+        Returns:
+            str: The generated closing message.
+        """
+
+        parameters_local = GenerationConfig(
+            temperature=0.7,
+            max_output_tokens=int(self.config["genai_chat"]["max_output_tokens"]),
+        )
+        chat = self.model.start_chat()
+        response = chat.send_message(
+            """You are Generative AI powered genai Learning Assistant.
+                Write a brief closing thank you message:"""
+        )
+        response = chat.send_message(f"""{text}""", generation_config=parameters_local)
+        message = response.text
+        return message
+
+    def genai_classify_intent(self, text: str) -> str:
+        """
+        Classifies the intent of an incoming message using a strict intent classifier model.
+
+        The supported intents are:
+            * 'WELCOME'
+            * 'WRITE_CODE'
+            * 'PROGRAMMING_QUESTION_AND_ANSWER'
+            * 'CLOSE'
+            * 'OTHER'
+            * 'FOLLOWUP'
+
+        Args:
+            text (str): The user's message.
+
+        Returns:
+            str: The classified intent.
+        """
+
+        response = self.model.generate_content(
+            f"""
```

Comment:
For code readability, I recommend putting prompts in a dedicated file.

Reply:
created a prompt_library.py file and added this prompt in the same file.